crates/engine/tree/src/tree/mod.rs

```diff
@@ -538,9 +543,11 @@ where
         persistence: PersistenceHandle,
         persistence_state: PersistenceState,
         payload_builder: PayloadBuilderHandle<T>,
+        state_root_task_spawner: Box<dyn TaskSpawner>,
         config: TreeConfig,
     ) -> Self {
         let (incoming_tx, incoming) = std::sync::mpsc::channel();
+        let blocking_task_pool = BlockingTaskPool::build().unwrap();
```

Comment:
let's propagate the error here?

Reply:
sure done here https://github.com/paradigmxyz/reth/pull/10927/commits/70ab8e394aa5b694e15e71f11936b011fd87e504

Reply:
hmm not sure, this would mean we can't build a rayon's thread pool, i've turned the unwrap into an expect like here https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/core.rs#L101